platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java

```diff
@@ -0,0 +1,67 @@
+package com.mapbox.mapboxsdk.maps;
+
+import static com.mapbox.mapboxsdk.maps.MapboxMap.OnCameraIdleListener;
+import static com.mapbox.mapboxsdk.maps.MapboxMap.OnCameraMoveCanceledListener;
+import static com.mapbox.mapboxsdk.maps.MapboxMap.OnCameraMoveStartedListener;
+import static com.mapbox.mapboxsdk.maps.MapboxMap.OnCameraMoveListener;
+
+class CameraChangeDispatcher implements MapboxMap.OnCameraMoveStartedListener, MapboxMap.OnCameraMoveListener,
+  MapboxMap.OnCameraMoveCanceledListener, OnCameraIdleListener {
+
+  private boolean idle = true;
+
+  private OnCameraMoveStartedListener onCameraMoveStartedListener;
+  private OnCameraMoveCanceledListener onCameraMoveCanceledListener;
+  private OnCameraMoveListener onCameraMoveListener;
+  private OnCameraIdleListener onCameraIdleListener;
+
+  void setOnCameraMoveStartedListener(OnCameraMoveStartedListener onCameraMoveStartedListener) {
+    this.onCameraMoveStartedListener = onCameraMoveStartedListener;
+  }
+
+  void setOnCameraMoveCanceledListener(OnCameraMoveCanceledListener onCameraMoveCanceledListener) {
+    this.onCameraMoveCanceledListener = onCameraMoveCanceledListener;
+  }
+
+  void setOnCameraMoveListener(OnCameraMoveListener onCameraMoveListener) {
+    this.onCameraMoveListener = onCameraMoveListener;
+  }
+
+  void setOnCameraIdleListener(OnCameraIdleListener onCameraIdleListener) {
+    this.onCameraIdleListener = onCameraIdleListener;
+  }
+
+  @Override
+  public void onCameraMoveStarted(int reason) {
+    if (!idle) {
```

Comment:
ah, checkstyle doesn't approve this.. `[ant:checkstyle] [ERROR] /home/tvn/Mapbox/mapbox-gl-native/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java:36: 'if' construct must use '{}'s. [NeedBraces]`

Feel free to ticket out changing this setting and doing a cleanup of the codebase.